drone/src/agent/executor.rs

```diff
@@ -262,16 +262,30 @@ impl Executor {
     ) -> Result<Option<BackendState>> {
         match state {
             BackendState::Loading => {
-                self.docker
-                    .pull_image(
-                        &spawn_request.executable.image,
-                        &spawn_request
-                            .executable
-                            .credentials
-                            .as_ref()
-                            .map(|d| d.into()),
-                    )
-                    .await?;
+                let should_pull_image;
+                match spawn_request.executable.pull_policy {
```

Comment:
match expressions can return a value! Instead of declaring should_pull_image and then setting it, you can set it to the value of the match expression